src/app/components/selectbutton/selectbutton.ts

```diff
@@ -159,7 +159,7 @@ export class SelectButton implements ControlValueAccessor {
         let optionValue = this.getOptionValue(option);
 
         if (this.multiple) {
-            if (this.value) {
+            if (this.value && this.value instanceof Array) {
```

Comment:
Good to know thanks :D Your solution looks way better. Mine only was a quick fix for the current problem

Reply:
Hello @volvachev. I tried your solution in my application. Works fine until I pass new options to the selectButton inside a dialog. Then I need the check for Array.isArray in isSelected also otherwise I have the same behavior.